Build fail Varnish7.4.1

```
$ make
make  all-recursive
make[1]: Entering directory '/home/ubuntu/tmp/libvmod-memcached'
Making all in src
make[2]: Entering directory '/home/ubuntu/tmp/libvmod-memcached/src'
/usr/share/varnish/vmodtool.py ../src/vmod_memcached.vcc

NOTICE: Please put $Module description in quotes.

/bin/bash ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I/usr/include/varnish -Wall -Werror  -I/usr/include/varnish  -g -O2 -MT libvmod_memcached_la-vmod_memcached.lo -MD -MP -MF .deps/libvmod_memcached_la-vmod_memcached.Tpo -c -o libvmod_memcached_la-vmod_memcached.lo `test -f 'vmod_memcached.c' || echo './'`vmod_memcached.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I/usr/include/varnish -Wall -Werror -I/usr/include/varnish -g -O2 -MT libvmod_memcached_la-vmod_memcached.lo -MD -MP -MF .deps/libvmod_memcached_la-vmod_memcached.Tpo -c vmod_memcached.c  -fPIC -DPIC -o .libs/libvmod_memcached_la-vmod_memcached.o
vmod_memcached.c: In function 'vmod_event':
vmod_memcached.c:65:21: error: 'struct vmod_priv' has no member named 'free'
   65 |                 priv->free = free_mc_vcl_settings;
      |                     ^~
vmod_memcached.c: In function 'vmod_servers':
vmod_memcached.c:118:32: error: incompatible type for argument 2 of 'VSL'
  118 |                 VSL(SLT_Debug, 0, "memcached pool config '%s'", config);
      |                                ^
      |                                |
      |                                int
In file included from vmod_memcached.c:10:
/usr/include/varnish/cache/cache.h:775:37: note: expected 'vxid_t' {aka 'struct vxids'} but argument is of type 'int'
  775 | void VSL(enum VSL_tag_e tag, vxid_t vxid, const char *fmt, ...)
      |                              ~~~~~~~^~~~
vmod_memcached.c:128:32: error: incompatible type for argument 2 of 'VSL'
  128 |                 VSL(SLT_Debug, 0, "memcached pool config '%s'", pool_str);
      |                                ^
      |                                |
      |                                int
In file included from vmod_memcached.c:10:
/usr/include/varnish/cache/cache.h:775:37: note: expected 'vxid_t' {aka 'struct vxids'} but argument is of type 'int'
  775 | void VSL(enum VSL_tag_e tag, vxid_t vxid, const char *fmt, ...)
      |                              ~~~~~~~^~~~
vmod_memcached.c:136:32: error: incompatible type for argument 2 of 'VSL'
  136 |                 VSL(SLT_Error, 0, "memcached servers() error");
      |                                ^
      |                                |
      |                                int
In file included from vmod_memcached.c:10:
/usr/include/varnish/cache/cache.h:775:37: note: expected 'vxid_t' {aka 'struct vxids'} but argument is of type 'int'
  775 | void VSL(enum VSL_tag_e tag, vxid_t vxid, const char *fmt, ...)
      |                              ~~~~~~~^~~~
vmod_memcached.c:137:32: error: incompatible type for argument 2 of 'VSL'
  137 |                 VSL(SLT_Error, 0, "%s", error_buf);
      |                                ^
      |                                |
      |                                int
In file included from vmod_memcached.c:10:
/usr/include/varnish/cache/cache.h:775:37: note: expected 'vxid_t' {aka 'struct vxids'} but argument is of type 'int'
  775 | void VSL(enum VSL_tag_e tag, vxid_t vxid, const char *fmt, ...)
      |                              ~~~~~~~^~~~
make[2]: *** [Makefile:479: libvmod_memcached_la-vmod_memcached.lo] Error 1
make[2]: Leaving directory '/home/ubuntu/tmp/libvmod-memcached/src'
make[1]: *** [Makefile:517: all-recursive] Error 1
make[1]: Leaving directory '/home/ubuntu/tmp/libvmod-memcached'
make: *** [Makefile:385: all] Error 2
```

It is not compatible with some Varnish changes.


# Enviroment

```
varnishd (varnish-7.4.1 revision d5a5aa9cc879320840ca467ddbb7df0f99c9ba0f)
libvmod-memcached 3163da9894e5656b0b47b1d7610e5ee4fbf7efd3
```

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -15,7 +15,9 @@ nodist_libvmod_memcached_la_SOURCES = \
 
 $(libvmod_memcached_la_OBJECTS): vcc_if.h
 
-vcc_if.c vcc_if.h: @VMODTOOL@ $(top_srcdir)/src/vmod_memcached.vcc
+vcc_if.c: vcc_if.h
+
+vcc_if.h: @VMODTOOL@ $(top_srcdir)/src/vmod_memcached.vcc
 	@VMODTOOL@ $(top_srcdir)/src/vmod_memcached.vcc
 
 VMOD_TESTS = $(top_srcdir)/src/tests/*.vtc

--- a/src/vmod_memcached.c
+++ b/src/vmod_memcached.c
@@ -38,7 +38,7 @@ struct vmod_mc_vcl_settings {
 };
 
 static void
-free_mc_vcl_settings(void *data)
+free_mc_vcl_settings(VRT_CTX, void *data)
 {
 	struct vmod_mc_vcl_settings *settings;
 	CAST_OBJ_NOTNULL(settings, data, VMOD_MC_SETTINGS_MAGIC);
@@ -46,6 +46,12 @@ free_mc_vcl_settings(void *data)
 	memcached_pool_destroy(settings->pool);
 	FREE_OBJ(settings);
 }
+
+static const struct vmod_priv_methods priv_vcl_methods[1] = {{
+	.magic = VMOD_PRIV_METHODS_MAGIC,
+	.type = "vmod_memcached_priv_vcl",
+	.fini = free_mc_vcl_settings
+}};
 
 int
 vmod_event(const struct vrt_ctx *ctx, struct vmod_priv *priv,
@@ -62,7 +68,7 @@ vmod_event(const struct vrt_ctx *ctx, struct vmod_priv *priv,
 		settings->error_str = POOL_ERROR_STRING;
 
 		priv->priv = settings;
-		priv->free = free_mc_vcl_settings;
+		priv->methods = priv_vcl_methods;
 	}
 
 	return (0);

--- a/src/vmod_memcached.c
+++ b/src/vmod_memcached.c
@@ -121,7 +121,7 @@ vmod_servers(const struct vrt_ctx *ctx, struct vmod_priv *priv,
 
 	if (strcasestr(config, POOL_MAX_CONN_PREFIX)) {
 		settings->pool = memcached_pool(config, strlen(config));
-		VSL(SLT_Debug, 0, "memcached pool config '%s'", config);
+		VSL(SLT_Debug, NO_VXID, "memcached pool config '%s'", config);
 	} else {
 		size_t pool_len = strlen(config) + strlen(POOL_MAX_CONN_PARAM);
 		char *pool_str = malloc(pool_len + 1);
@@ -131,7 +131,7 @@ vmod_servers(const struct vrt_ctx *ctx, struct vmod_priv *priv,
 
 		settings->pool = memcached_pool(pool_str, pool_len);
 
-		VSL(SLT_Debug, 0, "memcached pool config '%s'", pool_str);
+		VSL(SLT_Debug, NO_VXID, "memcached pool config '%s'", pool_str);
 
 		free(pool_str);
 	}
@@ -139,8 +139,8 @@ vmod_servers(const struct vrt_ctx *ctx, struct vmod_priv *priv,
 	if (!settings->pool) {
 		libmemcached_check_configuration(config, strlen(config),
 		    error_buf, sizeof(error_buf));
-		VSL(SLT_Error, 0, "memcached servers() error");
-		VSL(SLT_Error, 0, "%s", error_buf);
+		VSL(SLT_Error, NO_VXID, "memcached servers() error");
+		VSL(SLT_Error, NO_VXID, "%s", error_buf);
 	}
 }
 

--- a/src/vmod_memcached.vcc
+++ b/src/vmod_memcached.vcc
@@ -1,4 +1,4 @@
-$Module memcached 3 Varnish memcached module
+$Module memcached 3 "Varnish memcached module"
 $Event event
 $Function VOID servers(PRIV_VCL, STRING)
 $Function VOID error_string(PRIV_VCL, STRING)

